contents/tutorials/track-high-volume-apis.md

```diff
@@ -0,0 +1,266 @@
+---
+title: How to track high-volume APIs
+date: 2023-05-02
+author: ["ian-vanagas"]
+showTitle: true
+sidebar: Docs
+featuredImage: ../images/tutorials/banners/tutorial-10.png
+tags: ['feature flags', 'configuration']
+---
+
+Tracking high-volume APIs is a balancing act. You want to keep them as efficient as possible, while still capturing the usage data to improve them. This tutorial aims to help you find this balance. In it, we build an API in Express, add PostHog, then implement two solutions for tracking:
+
+1. Sampling using feature flags
+2. Caching usage and batching capture requests
+
+After doing this, we also show the different ways you can calculate insights.
+
+## Building our API
+
+To start, we will build a basic API that captures and returns the ID we passed to it. We use [Express, the Node server framework](/tutorials/node-express-analytics), to do this
+
+Create a folder for our project, go into the folder, and initialize npm. When prompted, select all the default options.
+
+```bash
+mkdir high-volume
+cd high-volume
+npm init
+```
+
+Once done, install Express.
+
+```bash
+npm i express
+```
+
+In your `high-volume` folder, create a file named `index.js`, and set up a basic Express app.
+
+```node
+// index.js
+const express = require('express');
+const app = express();
+const port = 3000
+
+app.get('/', (req, res) => {
+  res.send('Hello World!');
+})
+
+app.listen(port, () => {
+  console.log(`App listening on http://localhost:${port}`)
+})
+```
+
+Lastly, create one more route as our high volume requests route. We set this route as  `/big/:id`. For now, it returns a message using the request ID.
+
+```node
+//...
+app.get('/big/:id', (req, res) => {
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+### Adding PostHog
+
+Next, we install and set up PostHog. 
+
+```bash
+npm i posthog-node
+```
+
+We set up the PostHog client with our project API key and instance address, both of which you can get from your [project settings](https://app.posthog.com/project/settings). We add a `client.capture()` call to our `/big` route. The capture call includes the route distinct ID and an event name. We also include a `firstRouteCalled` user property to use for a filter later. 
+
+```node
+const express = require('express');
+const { PostHog } = require('posthog-node');
+const app = express();
+const port = 3000
+
+const client = new PostHog(
+  '<ph_project_api_key>',
+  { host: '<ph_instance_address>'}
+)
+
+//...
+
+app.get('/big/:id', (req, res) => {
+  client.capture(
+    { 
+      event: 'big route',
+      distinctId: req.params.id,
+			properties: {
+        $set_once: { firstRouteCalled: req.url },   
+      },
+    },
+  );
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+When we go to the big route again, we capture an event that we can see in our PostHog instance.
```

Comment:
I think since it's the only route we refer to, we don't always need to refer to it as big
```suggestion
When we go to our route again, we capture an event that we can see in our PostHog instance.
```